flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1325,26 +1495,38 @@ def run_to_convergence(self, desired_datasets, max_iterations=20):
             one desired state to another if there are no bugs.
         """
         for _ in range(max_iterations):
-            self.run_convergence_step(
-                dataset_map_from_iterable(desired_datasets))
-            note(self.current_datasets())
-            if compare_dataset_states(
-                self.current_datasets(),
-                dataset_map_from_iterable(desired_datasets),
-            ):
-                break
+            try:
+                self.run_convergence_step(
+                    dataset_map_from_iterable(desired_datasets))
+                if compare_dataset_states(
+                    self.current_datasets(),
+                    dataset_map_from_iterable(desired_datasets),
+                ):
+                    break
+            except Exception:
+                # The real loop will just continue if there are errors
+                # so we do too. This will sometimes occur when
+                # `eventualy_consistent` is set, when we try to do
+                # actions that aren't valid in the current state.
+                pass
```

Comment:
:eyes: This will lead to unhelpful failures of this test. I think ideally it would record the outcome
of each iteration and use it in the else clause to format a message. Too large of a
change for this branch probably though.